### DISCUSSION TITLE
Manual runtime from API?

### DISCUSSION THREAD

**Discussion by pjv** [sebr/bhyve-home-assistant#138] [Q&A]
Hi @sebr,

First some info: My device is an XD 2-zone timer (model [24632](https://www.orbitonline.com/products/b-hyve-xd-smart-hose-timer?variant=41990794674344)). This device works great with the integration overall, although it appears that a couple things don't seem to be implemented for it. I don't know if any of the b-hyve hose end timers have built-in flow sensors but this one doesn't appear to - neither in the app nor via the integration is there any water volume data. The other thing that doesn't seem to work is calling the `bhyve.set_manual_preset_runtime` service; it doesn't do anything at all.

Second, a question: If I start a manual watering event, either by using the physical buttons on the timer or by using the integration's `bhyve.start_watering` service, is there no way to interrogate the API for the number of minutes the zone has been manually set to water for?

When I use the service, I can know how many minutes I sent in the call so have this data independently, but if I use the buttons on the device I don't currently have any way to tell how long it is set to run for which would be nice to put on my dashboard.

**Comment by krusej23** [comment]
I have tried the set manual preset runtime service and it didn't seem to set it. Hopefully this can be figured out because that would stop me from having to use Bhyve's trash app.
